apps/src/lib/node/ledger/ethereum_node/test_tools.rs

```diff
@@ -30,17 +30,35 @@ pub mod mock_oracle {
     use namada::types::ethereum_events::EthereumEvent;
     use tokio::sync::mpsc::UnboundedSender;
     use tokio::sync::oneshot::Sender;
+    use tokio::task::LocalSet;
 
-    pub async fn run_oracle(
+    pub fn run_oracle(
         _: &str,
         _: UnboundedSender<EthereumEvent>,
         abort: Sender<()>,
-    ) {
-        loop {
-            if abort.is_closed() {
-                return;
-            }
-        }
+    ) -> tokio::task::JoinHandle<()> {
+        tokio::task::spawn_blocking(move || {
+            let rt = tokio::runtime::Handle::current();
+            rt.block_on(async move {
+                LocalSet::new()
+                    .run_until(async move {
+                        tracing::info!(
+                            "Mock Ethereum event oracle is starting"
+                        );
+
+                        loop {
+                            if abort.is_closed() {
+                                break;
+                            }
```

Comment:
There is probably a better way to busy wait here (`tokio::time::sleep` inbetween `abort.is_closed()` checks?).

Reply:
Use this
https://docs.rs/tokio/latest/tokio/sync/oneshot/struct.Sender.html#method.poll_closed

Reply:
Another thing is that you probably don't need to use `LocalSet` in the mock `run_oracle`. Just do a regular `tokio::spawn`, and return the handle from that.
```rust
tokio::spawn(async move {
    tracing::info!(
        "Mock Ethereum event oracle is starting"
    );

    poll_fn(|cx| abort.poll_closed(cx)).await;

    tracing::info!(
        "Mock Ethereum event oracle is no longer running"
    );
})
```
This should probably work

Reply:
This works, had to change the `abort: Sender<()>` parameter to be `mut abort: Sender<()>` as `poll_closed` requires mutability on `abort`. Should we change the parameter on the real `run_oracle` as well to be consistent? I need to learn why we need to do this - since we were passing ownership of the `Sender<()>` struct already, I would have thought we should be able to mutate it if we wanted to without having to add an extra keyword.

Reply:
We can keep using `abort: Sender<()>` as a parameter and instead do

```rust
            let mut abort = abort;
            poll_fn(|cx| abort.poll_closed(cx)).await;
```

I guess the compiler just needs an indication we will mutate this variable. The caller gives us ownership so they don't need to worry about it.

Reply:
if you indicate a variable will be mutated, at the function signature level, it will have no consequence on the type of the function. for instance:

This function

```rust
fn do_the_thing(mut x: i32) -> i32 { x = 0; x }
```

Is equivalent to this function, type wise.

```rust
fn do_the_thang(x: i32) -> i32 { x }
```

Reply:
Oh interesting, so the `mut` is just for the convenience of the code in the function so we don't have to redeclare. To the caller it is still the same. I thought it would change the type, I'll change this back quickly to use `mut` in the params